src/meshapi/views/forms.py

```diff
@@ -244,12 +262,14 @@ def network_number_assignment(request):
         print(f"NN Request failed. Could not decode request: {e}")
         return Response({"detail": "Got incomplete request"}, status=status.HTTP_400_BAD_REQUEST)
 
-    acquire_lock_for_all_installs()
+    # This call is a bit out of order, but we need to do this here so that we acquire the lock
+    # before we call Install.objects.select_for_update().get() below in order to prevent deadlocks
+    free_nn = get_next_free_nn()
```

Comment:
This does so to lock as soon as we know the request is valid? Makes sense, because I don't think that `get_next_free_nn` has any side effects.